studio/pages/project/[ref]/functions/[functionSlug]/index.tsx

```diff
@@ -26,32 +29,95 @@ const CHART_INTERVALS: ChartIntervals[] = [
     startUnit: 'minute',
     format: 'MMM D, h:mm:ssa',
   },
-  { key: '15min', label: '15 min', startValue: 15, startUnit: 'minute', format: 'MMM D, h:mma' },
-  { key: '1hr', label: '1 hour', startValue: 1, startUnit: 'hour', format: 'MMM D, h:mma' },
-  { key: '1day', label: '1 day', startValue: 1, startUnit: 'hour', format: 'MMM D, h:mma' },
-  { key: '7day', label: '7 days', startValue: 7, startUnit: 'day', format: 'MMM D' },
+  {
+    key: '15min',
+    label: '15 min',
+    startValue: 15,
+    startUnit: 'minute',
+    format: 'MMM D, h:mma',
+  },
+  {
+    key: '1hr',
+    label: '1 hour',
+    startValue: 1,
+    startUnit: 'hour',
+    format: 'MMM D, h:mma',
+  },
+  {
+    key: '1day',
+    label: '1 day',
+    startValue: 1,
+    startUnit: 'hour',
+    format: 'MMM D, h:mma',
+  },
+  {
+    key: '7day',
+    label: '7 days',
+    startValue: 7,
+    startUnit: 'day',
+    format: 'MMM D',
+  },
 ]
 
 const PageLayout: NextPageWithLayout = () => {
   const router = useRouter()
   const { ref: projectRef, functionSlug } = useParams()
   const [interval, setInterval] = useState<string>('15min')
   const selectedInterval = CHART_INTERVALS.find((i) => i.key === interval) || CHART_INTERVALS[1]
-  const { data: selectedFunction } = useEdgeFunctionQuery({ projectRef, slug: functionSlug })
-  const id = selectedFunction?.id
-
-  const { data, error } = useFunctionsInvStatsQuery({
+  const { data: selectedFunction } = useEdgeFunctionQuery({
     projectRef,
-    functionId: id,
-    interval: selectedInterval.key,
+    slug: functionSlug,
   })
-  const isChartLoading = !data?.result && !error ? true : false
-  const normalizedData = useMemo(() => {
-    return (data?.result || []).map((d: any) => ({
+  const id = selectedFunction?.id
+  const resourceUsageMetricsEnabled = useFlag('enableResourceUsageMetricsForEdgeFunctions')
+
+  const { data: invStatsData, error: invStatsError } = useFunctionsInvStatsQuery(
+    {
+      projectRef,
+      functionId: id,
+      interval: selectedInterval.key,
+    },
+    { enabled: !resourceUsageMetricsEnabled }
+  )
+
+  const { data: reqStatsData, error: reqStatsError } = useFunctionsReqStatsQuery(
+    {
+      projectRef,
+      functionId: id,
+      interval: selectedInterval.key,
+    },
+    { enabled: resourceUsageMetricsEnabled }
+  )
+
+  const { data: resourceUsageData, error: resourceUsageError } = useFunctionsResourceUsageQuery(
+    {
+      projectRef,
+      functionId: id,
+      interval: selectedInterval.key,
+    },
+    { enabled: resourceUsageMetricsEnabled }
+  )
+
+  const isInvStatsChartLoading = !invStatsData || (!invStatsData.result && !invStatsError)
+  const isReqStatsChartLoading = !reqStatsData || (!reqStatsData?.result && !reqStatsError)
+  const isResourceUsageChartLoading =
+    !resourceUsageData || (!resourceUsageData?.result && !resourceUsageError)
+
+  const normalizedReqStatsData = useMemo(() => {
```

Comment:
this normalization function can actually be removed if the LF endpoint returns the `timestamp` field as a `DATETIME`. I don't think i got to refactoring and verifying that the invocation stats returns an iso timestamp, hence why it remained.
The new endpoints should use `select datetime(t.timestamp)` in the query

Reply:
removing these data normalization hooks would make the chart more performant as well

Reply:
ah cool, looks like the sql queries are already returning iso time stamps. removed normalization functions.